subprojects/signing/src/main/java/org/gradle/plugins/signing/SigningExtension.java

```diff
@@ -393,7 +386,10 @@ public List<Sign> sign(DomainObjectCollection<Publication> publications) {
     }
 
     private <T extends PublicationArtifact> Sign createSignTaskFor(final PublicationInternal<T> publicationToSign) {
-        final Function<String, Sign> computeSignTask = signTaskName -> {
+        final String signTaskName = determineSignTaskNameForPublication(publicationToSign);
+        if (project.getTasks().getNames().contains(signTaskName)) {
+            return project.getTasks().named(signTaskName, Sign.class).get();
```

Comment:
Note that this will change the error message when name collision occurs but the tasks have different types. But maybe we do not need to preserve that.

Reply:
Yes should I add a type check for the task instead of letting `named(String, Class)` throw an exception, and throw a more specific error message?